superset-frontend/packages/superset-ui-core/src/ui-overrides/ExtensionsRegistry.ts

```diff
@@ -109,6 +109,27 @@ export interface DatabaseConnectionExtension {
   onEdit?: (componentState: any) => void;
 }
 
+/**
+ * Interface for extensions SQL Form.
+ * These will be passed from the SQLEditor
+ *
+ * You can specify the types to be used for these properties
+ * for better type checking or use the default ones.
+ *
+ * @template TTables - Type for the tables array.
+ * @template TQueryEditor - Type for the query editor.
+ * @template TDatabase - Type for the database.
+ */
+export interface SQLFormExtensionProps<
+  TTables = any[],
+  TQueryEditor = object,
+  TDatabase = any,
+> {
+  tables: TTables[];
+  queryEditor: TQueryEditor;
+  database: TDatabase;
```

Comment:
@Antonio-RiveroMartnez 
Not all custom extension requires all these properties such like `tables` and `database`
Therefore, I think it's not a good idea passing these specified properties because you can access these values from redux. (you can also access the queryEditor object by useQueryEditor hook)
Only thing you need to pass is `queryEditorId` and then each custom extension component should use a hook to retrieve the properties that needed.

```suggestion
  queryEditorId: string;
```

Reply:
Hey @justinpark ! Good call, I will change this. I am working a bit more on this extension feature so I'll move this PR to be a DRAFT until I get everything in place and final props etc are figured out. ILYK. Thanks for the input as always 👏